pulsar-broker/src/main/java/org/apache/pulsar/broker/systopic/SystemTopicClient.java

```diff
@@ -168,12 +172,23 @@
     }
 
     static boolean isSystemTopic(TopicName topicName) {
-        if (topicName.isPartitioned()) {
-            return EventsTopicNames.NAMESPACE_EVENTS_LOCAL_NAME
-                    .equals(TopicName.get(topicName.getPartitionedTopicName()).getLocalName());
+        // system namespace
```

Comment:
I agree that the logic in this method is insufficient. I initially proposed a similar change here: https://github.com/apache/pulsar/pull/10254. In general, I think it would be valuable to take a closer look at how we manage system topics. It'd be great to have a standard way to define system topics so that we don't have to enumerate them in these types of checks, especially because I expect that we'll add more system topics over time.